### PR TITLE
Fix doc links in generated READMEs

### PR DESCRIPTION
**Pull request by stiiifff**

### CODE REVIEW

**Comment by stiiifff** [comment]
Relative links are rewritten to point to docs.rs. The logic is hard-coded in `commands/gen_readme/fix_doc_links` for now.
This could be made parameterized at a later point, if needed.

**Comment by gnunicorn** [comment]
@stiiifff awsome, thanks!

You think we could use [the documentation field from the Cargo Manifest](https://doc.rust-lang.org/cargo/reference/manifest.html#the-documentation-field) and only fallback to `docs.rs` if none found? Then I don't think we need to parameterize any further..

But for parity, there is some push to host our own version and point to that - and generally that seems a legit case, considering that the field exists.

**Comment by stiiifff** [comment]
Of course we could, are we setting the documentation field already ?

**Comment by gnunicorn** [comment]
I think we have/are.

**Comment by stiiifff** [comment]
@gnunicorn Added the use of the documentation field if defined (if not use docs.rs as default)
+ use of lazy_static to avoid initializing the regex for every package.

**Review by gnunicorn — APPROVED** [review]
LGTM

File: src/commands/readme.rs [hunk 1]
```diff
@@ -203,3 +214,25 @@ fn find_readme_template<'a>(
         None
     })
 }
+
+fn fix_doc_links(pkg_name: &str, readme: &str, doc_uri: Option<&str>) -> String {
```
Comment (Xanewok) [inline src/commands/readme.rs:218]: I don't think it's urgent but it'd be great to add some more documentation here - at a first glance it's not obvious what this functions is supposed to do - either at the function level or at the first two arms of the `match`

Comment (stiiifff) [reply src/commands/readme.rs:218]: Opened another PR w/ more readable code, fixes & tests: https://github.com/paritytech/cargo-unleash/pull/19

**Pull request merged**